watertap/property_models/activated_sludge/tests/test_asm2d_reaction.py

```diff
@@ -606,3 +610,303 @@ def test_solution(self, model):
+        EPS = 1e-8
+
+        # Feed conditions based on manual mass balance of inlet and recycle streams
+        m.fs.R1.inlet.flow_vol.fix(92230 * units.m**3 / units.day)
+        m.fs.R1.inlet.temperature.fix(298.15 * units.K)
+        m.fs.R1.inlet.pressure.fix(1 * units.atm)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_O2"].fix(7.9707 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_N2"].fix(29.0603 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_NH4"].fix(8.0209 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_NO3"].fix(6.6395 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_PO4"].fix(7.8953 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_F"].fix(0.4748 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_A"].fix(0.0336 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_I"].fix(30 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_I"].fix(1695.7695 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_S"].fix(68.2975 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_H"].fix(1855.5067 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PAO"].fix(214.5319 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PP"].fix(63.5316 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PHA"].fix(2.7381 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_AUT"].fix(118.3582 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_MeOH"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_MeP"].fix(EPS * units.mg / units.liter)
+        # No data on TSS from EXPOsan at this point
+        # However, TSS is needed for this reaction
+        m.fs.R1.inlet.conc_mass_comp[0, "X_TSS"].fix(3525.429 * units.mg / units.liter)
+
+        # Alkalinity was given in mg/L based on C
+        m.fs.R1.inlet.alkalinity[0].fix(4.6663 * units.mmol / units.liter)
+
+        m.fs.R1.volume.fix(1000 * units.m**3)
+
+        return m
+
+    @pytest.mark.component
+    def test_dof(self, model):
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model) == 0
+
+    @pytest.mark.component
+    def test_solve(self, model):
+        model.fs.R1.initialize(optarg={"bound_push": 1e-8, "mu_init": 1e-8})
+
+        solver = get_solver()
+        solver.options = {"bound_push": 1e-8, "mu_init": 1e-8}
+        results = solver.solve(model, tee=True)
+
+        assert check_optimal_termination(results)
+
+    @pytest.mark.component
+    def test_solution(self, model):
+        # EXPOsan calculations appear to be slightly off from this implementation
+        # It is supected that this is due to an error in the EXPOsan stoichiometric
+        # coefficient for alkalinity
```

Comment:
Applicable to your test, or just residual from older tests that could deleted?

Reply:
Residual, will delete